src/GraphQL/Mutation/MutationType.php

```diff
@@ -860,24 +871,18 @@ public function buildDeleteElementMutation(&$config, $context, $type)
 
             $omitPermissionCheck = $this->omitPermissionCheck;
 
+            $me = $this;
             $deleteField = [
                 'type' => $deleteResultType,
                 'args' => [
-                    'id' => ['type' => Type::nonNull(Type::int())],
-                ], 'resolve' => static function ($value, $args, $context, ResolveInfo $info) use ($type, $omitPermissionCheck) {
+                    'id' => ['type' => Type::int()],
+                    'fullpath' => ['type' => Type::string()],
+                ], 'resolve' => static function ($value, $args, $context, ResolveInfo $info) use ($type, $omitPermissionCheck, $me) {
                     try {
                         $id = $args["id"];
                         /** @var $configuration Configuration */
                         $configuration = $context['configuration'];
-                        $element = null;
-
-                        if ($type == "asset") {
-                            $element = Asset::getById($id);
-                        } else if ($type == "document") {
-                            $element = Document::getById($id);
-                        } else if ($type == "object") {
-                            $element = DataObject::getById($id);
-                        }
+                        $element = $me->getElementByTypeAndIdOrPath($args, $type);
 
```

Comment:
@weisswurstkanone I replaced all the different handling of asset, document and object by the new trait 'ElementIdentificationTrait' which gets the elements from the "standard Pimcore functions" Service::getElementById and Service::getElementByPath. 
That way we do not have to know implementation details from Pimcore and the Service will manage it for us.

Reply:
excellent. please let me know when it is ready to be reviewed

Reply:
A first draft of the PR is ready to be reviewed.
We do have a couple of questions though which would have to be answered first before we can finish up the entire functionality.